### PR TITLE
(Ledger store benchmark - 3/N) Add comments about the benchmark and its arguments

### PR DESCRIPTION
**Pull request by yhchiang-sol**
#### Summary of Changes
Add comments about the ledger store benchmark and its arguments.

This PR is based on top of #22091

### CODE REVIEW

File: core/tests/ledger_cleanup.rs [hunk 1]
```diff
@@ -151,6 +151,41 @@ mod tests {
...
+    ///   benchmark will stop immediately when the number of times where the
+    ///   ledger store size exceeds the configured `STOP_SIZE_BYTES`.  These
+    ///   configs are used to make sure the benchmark runs successfully under
+    ///   the storage limitation.
+    /// - `CLEANUP_BLOCKSTORE`: if true, the ledger store created in the current
+    ///   benchmark run will be deleted.  Default: true.
+    /// - `NO_COMPACTION`: whether to stop the rocksdb's background compaction
+    ///   completely.  Default: false.
+    /// - `ASSERT_COMPACTION`: if true, any RocksDB compaction will hit an
+    ///   assertion.  Default: false.
```
Comment (steviez) [inline core/tests/ledger_cleanup.rs:180]: nits:
- Consistency on `rocksdb` vs. `RocksDB`
- Do you think these two settings should be moved under the `Cleanup-service related settings` section too?
- `ASSERT_COMPACITON` can't explicitly be tied to compaction by Rocks or LedgerCleanupService; rather, the assertion simply checks if the storage size shrunk

Comment (yhchiang-sol) [reply core/tests/ledger_cleanup.rs:180]: Thanks for the correction.  Will correct the comment to reflect that the code does.

Comment (yhchiang-sol) [reply core/tests/ledger_cleanup.rs:180]: > Do you think these two settings should be moved under the Cleanup-service related settings section too?

I think `ASSERT_COMPACTION` should go under the cleanup service, while NO_COMPACTION should be independent of the cleanup service, as we can run a benchmark with neither cleanup service nor compaction to see how fast we can do without them.

File: core/tests/ledger_cleanup.rs [hunk 2]
```diff
@@ -151,6 +151,42 @@ mod tests {
...
+    /// - `STOP_SIZE_BYTES`: if specified, the benchmark will count how
+    ///   many times the ledger store size exceeds the specified threshold.
+    /// - `STOP_SIZE_ITERATIONS`: when `STOP_SIZE_BYTES` is specified, the
+    ///   benchmark will stop immediately when the number of times where the
+    ///   ledger store size exceeds the configured `STOP_SIZE_BYTES`.  These
+    ///   configs are used to make sure the benchmark runs successfully under
+    ///   the storage limitation.
+    /// - `CLEANUP_BLOCKSTORE`: if true, the ledger store created in the current
+    ///   benchmark run will be deleted.  Default: true.
+    /// - `NO_COMPACTION`: whether to stop the rocksdb's background compaction
```
Comment (steviez) [inline core/tests/ledger_cleanup.rs:177]: nit: "whether to stop ~~the~~ rocksdb's ..."

File: core/tests/ledger_cleanup.rs [hunk 3]
```diff
@@ -151,6 +151,42 @@ mod tests {
...
+    ///   completely.  Default: false.
+    ///
+    /// Cleanup-service related settings:
+    /// - `MAX_LEDGER_SHREDS`: when the clean-up service is on, the service will
+    ///   clean up the ledger store when the number of shreds exceed this value.
```
Comment (steviez) [inline core/tests/ledger_cleanup.rs:182]: nit: "number of shreds ~~exceed~~ exceeds this value."

File: core/tests/ledger_cleanup.rs [hunk 4]
```diff
@@ -151,6 +151,42 @@ mod tests {
...
+    /// - `COMPACTION_INTERVAL`: if set, the clean-up service will compact all
+    ///   slots which is older than the specified interval.  The interval is
```
Comment (steviez) [inline core/tests/ledger_cleanup.rs:184]: nit: "slots ~~which is~~ that are older than ..."

File: core/tests/ledger_cleanup.rs [hunk 5]
```diff
@@ -151,6 +151,42 @@ mod tests {
...
+    ///   measured by slots.
+    ///   Default: the number of slots per day (`TICKS_PER_DAY` / `DEFAULT_TICKS_PER_SLOT`)
+    /// - `ASSERT_COMPACTION`: if true, then it the benchmark will perform a sanity
```
Comment (steviez) [inline core/tests/ledger_cleanup.rs:187]: nit: "if true, then ~~it~~ the benchmark ..."

**Review by steviez — APPROVED** [review]
Just some minor grammar mistakes; looks good to me after you fix those though!

**Pull request merged**